app/app.js

```diff
@@ -70,108 +74,16 @@ module.exports = (options) => {
-      examplesDirectory: examples,
-      fullPageExamplesDirectory: fullPageExamples
-    })
-  })
-
-  // Whenever the route includes a :component parameter, read the component data
-  // from its YAML file
-  app.param('component', function (req, res, next, componentName) {
-    res.locals.componentData = fileHelper.getComponentData(componentName)
-    next()
-  })
+  app.use('/', indexRouter)
 
   // All components view
-  app.get('/components/all', function (req, res, next) {
-    const components = fileHelper.allComponents
-
-    res.locals.componentData = components.map(componentName => {
-      let componentData = fileHelper.getComponentData(componentName)
-      let defaultExample = componentData.examples.find(
-        example => example.name === 'default'
-      )
-      return {
-        componentName,
-        examples: [defaultExample]
-      }
-    })
-    res.render(`all-components`, function (error, html) {
-      if (error) {
-        next(error)
-      } else {
-        res.send(html)
-      }
-    })
-  })
-
-  // Component 'README' page
-  app.get('/components/:component', function (req, res, next) {
-    // make variables available to nunjucks template
-    res.locals.componentPath = req.params.component
-
-    res.render('component', function (error, html) {
-      if (error) {
-        next(error)
-      } else {
-        res.send(html)
-      }
-    })
-  })
-
-  // Component example preview
-  app.get('/components/:component/:example*?/preview', function (req, res, next) {
-    // Find the data for the specified example (or the default example)
-    let componentName = req.params.component
-    let requestedExampleName = req.params.example || 'default'
-
-    let previewLayout = res.locals.componentData.previewLayout || 'layout'
-
-    let exampleConfig = res.locals.componentData.examples.find(
-      example => example.name.replace(/ /g, '-') === requestedExampleName
-    )
-
-    if (!exampleConfig) {
-      next()
-    }
-
-    // Construct and evaluate the component with the data for this example
-    let macroName = helperFunctions.componentNameToMacroName(componentName)
-    let macroParameters = JSON.stringify(exampleConfig.data, null, '\t')
-
-    res.locals.componentView = env.renderString(
-      `{% from '${componentName}/macro.njk' import ${macroName} %}
-      {{ ${macroName}(${macroParameters}) }}`
-    )
-
-    let bodyClasses = ''
-    if (req.query.iframe) {
-      bodyClasses = 'app-iframe-in-component-preview'
-    }
-
-    res.render('component-preview', { bodyClasses, previewLayout })
-  })
+  app.use('/components', componentRouter)
 
   // Example view
-  app.get('/examples/:example', function (req, res, next) {
-    res.render(`${req.params.example}/index`, function (error, html) {
-      if (error) {
-        next(error)
-      } else {
-        res.send(html)
-      }
-    })
-  })
+  app.use('/examples', exampleRouter)
 
   // Full page example views
-  require('./full-page-examples.js')(app)
```

Comment:
This line confused me because it wasn't clear what it was doing. I had to go into  `full-page-examples.js` only to find some more weirdness with requiring more files ( oddly not all the examples in `full-page-examples` which were required) for post requests and form validations.

Instead I copied the route pattern we have used before but in the Route file it adds controllers for the forms we are interested in.

Reply:
I'll also update https://github.com/alphagov/govuk-frontend/pull/1274/files#diff-2de9eda1655d499e934e7465e87ca0bdL68 so it follows this new pattern